src/retime.jl

```diff
@@ -152,13 +157,23 @@ function _retime!(
     return nothing
 end
 
-function __get_new_values(T, N, n, extrapolate, skip_missing)
-    return zeros(skip_missing ? nonmissingtype(T) : T, N, n)
-end
-function __get_new_values(T, N, n, extrapolate::MissingExtrapolate, skip_missing)
-    return zeros(Union{Missing,T}, N, n)
+function __allocate_new_values(T, N, n, upsample, downsample, extrapolate, skip_missing)
+    T = skip_missing ? nonmissingtype(T) : T
+    new_type = promote_type(
+        T, __get_type(T, upsample), __get_type(T, downsample), __get_type(T, extrapolate)
+    )
+    return zeros(new_type, N, n)
```

Comment:
ah, here is an issue about `skip_missing`:

```julia
julia> typeof(TimeSeries.__allocate_new_values(Float64, 42, 4, Linear(), Mean(), NearestExtrapolate(), false))
Matrix{Float64} (alias for Array{Float64, 2})

julia> typeof(TimeSeries.__allocate_new_values(Float64, 42, 4, Linear(), Mean(), NearestExtrapolate(), true))
Matrix{Float64} (alias for Array{Float64, 2})
```

Could you add a test case for this issue?